server/plugin/virtual_agent_test.go

```diff
@@ -452,3 +455,120 @@ func Test_CreateMessageAttachment(t *testing.T) {
+		},
+	} {
+		t.Run(test.description, func(t *testing.T) {
+			defer monkey.UnpatchAll()
+			mockAPI := &plugintest.API{}
+			test.setupPlugin(&p, mockAPI)
+			p.SetAPI(mockAPI)
+			defer mockAPI.AssertExpectations(t)
+			mockCtrl := gomock.NewController(t)
+			mockedStore := mock_plugin.NewMockStore(mockCtrl)
+			test.setupStore(mockedStore)
+			p.store = mockedStore
+
+			err := p.HandleCarouselInput(testutils.GetID(), testutils.GetPickerBodyWithCarouselOptions(3))
+			if test.expectedError == nil {
+				require.Nil(t, err)
+			} else {
+				assert.EqualError(t, err, test.expectedError.Error())
+			}
+		})
+	}
+}
+
+func Test_HandlePreviousCarouselPosts(t *testing.T) {
+	p := Plugin{}
+	for _, test := range []struct {
+		description string
+		setupAPI    func(api *plugintest.API)
+		setupStore  func(s *mock_plugin.MockStore)
+	}{
+		{
+			description: "error in loading postIDs",
+			setupAPI: func(api *plugintest.API) {
+				api.On("LogDebug", testutils.GetMockArgumentsWithType("string", 5)...).Return()
+			},
+			setupStore: func(s *mock_plugin.MockStore) {
+				s.EXPECT().LoadPostIDs(testutils.GetID()).Return(nil, errors.New("error in loading post IDs"))
+			},
+		},
+		{
+			description: "no postIDs present in KV store",
+			setupAPI:    func(api *plugintest.API) {},
+			setupStore: func(s *mock_plugin.MockStore) {
+				s.EXPECT().LoadPostIDs(testutils.GetID()).Return([]string{}, nil)
+			},
+		},
+		{
+			description: "error in storing postIDs and getting post from API",
+			setupAPI: func(api *plugintest.API) {
+				api.On("LogDebug", testutils.GetMockArgumentsWithType("string", 5)...).Return()
+				api.On("GetPost", testutils.GetID()).Return(nil, testutils.GetAppError("error in getting post"))
+			},
+			setupStore: func(s *mock_plugin.MockStore) {
+				s.EXPECT().LoadPostIDs(testutils.GetID()).Return([]string{testutils.GetID()}, nil)
+				s.EXPECT().StorePostIDs(testutils.GetID(), []string{}).Return(errors.New("error in storing post IDs"))
+			},
+		},
+		{
+			description: "error in updating post",
+			setupAPI: func(api *plugintest.API) {
+				api.On("GetPost", testutils.GetID()).Return(testutils.GetPostWithAttachments(2), nil)
+				api.On("UpdatePost", mock.AnythingOfType("*model.Post")).Return(nil, testutils.GetAppError("error in updating post"))
+				api.On("LogDebug", testutils.GetMockArgumentsWithType("string", 5)...).Return()
+			},
+			setupStore: func(s *mock_plugin.MockStore) {
+				s.EXPECT().LoadPostIDs(testutils.GetID()).Return([]string{testutils.GetID()}, nil)
+				s.EXPECT().StorePostIDs(testutils.GetID(), []string{}).Return(nil)
+			},
+		},
+	} {
+		t.Run(test.description, func(t *testing.T) {
+			mockAPI := &plugintest.API{}
+			test.setupAPI(mockAPI)
+			p.SetAPI(mockAPI)
+			mockCtrl := gomock.NewController(t)
+			mockedStore := mock_plugin.NewMockStore(mockCtrl)
+			test.setupStore(mockedStore)
+			p.store = mockedStore
+
+			p.handlePreviousCarouselPosts(testutils.GetID())
```

Comment:
don't we need to check something here?

Reply:
No, we don't. If you see the function definition, there's nothing to check. Besides, all the assertions are already happening when we are calling `Expect` function of the mocks.